stdlib/public/Concurrency/CMakeLists.txt

```diff
@@ -199,6 +199,8 @@ if(SWIFT_SHOULD_BUILD_EMBEDDED_STDLIB AND SWIFT_SHOULD_BUILD_EMBEDDED_CONCURRENC
   set(SWIFT_CONCURRENCY_USES_DISPATCH FALSE)
   set(SWIFT_STDLIB_SINGLE_THREADED_CONCURRENCY TRUE)
```

Comment:
Shouldn't this change? We should want to eventually get to being able to specify `SWIFT_CONCURRENCY_GLOBAL_EXECUTOR_default = hooked` so that `SWIFT_CONCURRENCY_GLOBAL_EXECUTOR` is set to hooked.

Reply:
Let's do it as a follow-up. If I just naively remove this line, I get...
```
Undefined symbols for architecture arm64:
  "_swift_reportError", referenced from:
      _swift_task_enqueueGlobal in libswift_Concurrency.a[7](GlobalExecutor.cpp.o)
      swift_task_enqueueGlobalImpl(swift::Job*) in libswift_Concurrency.a[7](GlobalExecutor.cpp.o)
      swift_task_asyncMainDrainQueueImpl() in libswift_Concurrency.a[11](Task.cpp.o)
ld: symbol(s) not found for architecture arm64
```
...because the hooked version (`SWIFT_CONCURRENCY_GLOBAL_EXECUTOR_default = hooked`) of Concurrency still brings in the stub implementations from NonDispatchGlobalExecutor.inc, and we should instead really just have these symbols undefined in the runtime, and expect the user to provide them and link them.

Reply:
>  and we should instead really just have these symbols undefined in the runtime, and expect the user to provide them and link them.

Don't we just want to funnel this to swift_fatal_error?

Reply:
That seems like the wrong solution -- it would certainly work, but we are going to need to keep cutting down dependencies and code in the Embedded Swift Concurrency runtime, and IIUC, in hooked mode, we expect the user/platform to provide `swift_task_enqueueGlobal`, and therefore the two symbols from the listing above (`_swift_task_enqueueGlobal`, `swift_task_enqueueGlobalImpl `) should really just not be part of the hooked build of Concurrency. Doing that sounds like a double win (besides resolving this undefined symbol problem): (1) less code = better codesize, (2) turns a runtime failure into a link time error.